File: pytools/data/data.code.golang.server/req_scene_enter.go
package proto

import (
	"packet"
)

type ReqSceneEnter struct {
	doorId                   uint32
}

func ReqSceneEnterDecode(pack *packet.Packet) *ReqSceneEnter {
	reqSceneEnter := &ReqSceneEnter{}

	reqSceneEnter.doorId = pack.ReadUint32()
	return reqSceneEnter
}

func (this *ReqSceneEnter) GetDoorId() uint32 {
	return this.doorId
}

File: pytools/examples/lua/server/protocols/ReqSceneEnter.lua
module("ReqSceneEnter", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqSceneEnter})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.door_id = 0
end


function decode(self, pack)
	self.door_id = pack:readU32()
	return self
end


function getDoorId(self)
	return self.door_id
end

